source/common/http/http1/codec_impl.cc

```diff
@@ -411,8 +411,9 @@ int ServerConnectionImpl::onHeadersComplete(HeaderMapImplPtr&& headers) {
         http_parser_pause(&parser_, 1);
       }
 
-    } else if (parser_.method == http_method::HTTP_POST ||
-               parser_.method == http_method::HTTP_PUT) {
+    } else if ((parser_.method == http_method::HTTP_POST ||
+                parser_.method == http_method::HTTP_PUT) &&
+               headers->ContentLength() == nullptr) {
```

Comment:
Yup happy to chat. In general I would put a change like this behind "runtime" control or doing a stat first, but not sure if it's worth doing all that extra work if we can feel comfortable with it.

However, I'm still honestly not sure this is correct or we might need to expand. The reason is that AFAIK in the new RFC, content-length is totally decoupled from method. So, I think for HTTP/1.1, we need to rely on caller actually ending the HTTP/1.1 message and potentially not do this check? See comment here: https://github.com/lyft/envoy/blob/master/source/common/http/http1/codec_impl.cc#L398